packages/langium/src/grammar/generated/ast.ts

```diff
@@ -200,7 +201,7 @@ export function isInferredType(item: unknown): item is InferredType {
 }
 
 export interface Interface extends AstNode {
-    readonly $container: Grammar;
+    readonly $container: Alternatives | Assignment | AtomType | CharacterRange | CrossReference | Grammar | Group | NegatedToken | ParserRule | TerminalAlternatives | TerminalGroup | TerminalRule | UnorderedGroup | UntilToken;
```

Comment:
This doesn't seem to be right. 
Can e.g. an assignment be a container of an Interface?  Other `$container` types also doesn't seem to be right to me.

Reply:
True, it's not right by the fact: not all listed types are real `$container`s of `Interface`. However, we have them here since we collect `$container` types per connected component and not per type. We have to do it because one type can inherit some types and having `$container` types per type brings TS errors. Try an example:
```
interface A { $container: D; strA: string }
interface B { $container: E; strB: string }
interface C extends A, B { $container: D | E; strC: string }
interface D { a: A }
interface E { b: B }
```
As the result, our solution is having `$container: D | E` everywhere: for `A` and `B` to. It's not a perfect solution, of course, so, we have an issue about improving `$container` types calculation (#150). Current PR doesn't solve the problem, but adds `$container` types for declared types (#750).

Reply:
But `Interface` is completely isolated in it's inheritance structure. Why did it change anyway?

Reply:
Well okay, when the issue is already addressed

Reply:
@msujew , `Interface` is not isolated: we have `type AbstractType = Interface | Type | Action | ParserRule;` that connects it with `ParserRule` that is connected with `AbstractElement` -- what gives so many container types for each participant of this hierarchy.